connectivity/check/deployment.go

```diff
@@ -389,24 +389,28 @@ func newIngress() *networkingv1.Ingress {
 
 // deploy ensures the test Namespace, Services and Deployments are running on the cluster.
 func (ct *ConnectivityTest) deploy(ctx context.Context) error {
-	if ct.params.ForceDeploy {
-		if err := ct.deleteDeployments(ctx, ct.clients.src); err != nil {
-			return err
-		}
-	}
+	var err error
 
-	_, err := ct.clients.src.GetNamespace(ctx, ct.params.TestNamespace, metav1.GetOptions{})
-	if err != nil {
-		ct.Logf("✨ [%s] Creating namespace %s for connectivity check...", ct.clients.src.ClusterName(), ct.params.TestNamespace)
-		namespace := &corev1.Namespace{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:        ct.params.TestNamespace,
-				Annotations: ct.params.NamespaceAnnotations,
-			},
+	for _, client := range ct.Clients() {
+		if ct.params.ForceDeploy {
+			if err := ct.deleteDeployments(ctx, client); err != nil {
+				return err
+			}
 		}
-		_, err = ct.clients.src.CreateNamespace(ctx, namespace, metav1.CreateOptions{})
+
+		_, err := client.GetNamespace(ctx, ct.params.TestNamespace, metav1.GetOptions{})
 		if err != nil {
-			return fmt.Errorf("unable to create namespace %s: %s", ct.params.TestNamespace, err)
+			ct.Logf("✨ [%s] Creating namespace %s for connectivity check...", client.ClusterName(), ct.params.TestNamespace)
+			namespace := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        ct.params.TestNamespace,
+					Annotations: ct.params.NamespaceAnnotations,
+				},
+			}
+			_, err = client.CreateNamespace(ctx, namespace, metav1.CreateOptions{})
+			if err != nil {
+				return fmt.Errorf("unable to create namespace %s: %s", ct.params.TestNamespace, err)
```

Comment:
This line and the error handling for the Service creation (line 643) should be made similar. A message could be added below and `%w` could be used for the err format specifier.